### PR TITLE
flecks (tm)

### PR DESCRIPTION
**Pull request by Vlue**
rai made me do it

### CODE REVIEW

File: src/main/resources/assets/trickster/lavender/entries/tome_of_tomfoolery/ploys/flecks.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+```json
+{
+  "title": "Flecks",
+  "icon": "minecraft:ghast_tear",
+  "category": "trickster:ploys"
+}
+```
+
+Flecks are ... and can be optionally only shown to a given player or list of players. all flecks return their id, for convenience
```
Comment (StellarWitch7) [inline src/main/resources/assets/trickster/lavender/entries/tome_of_tomfoolery/ploys/flecks.md:9]: Sentence needs polishing.

File: src/main/java/dev/enjarai/trickster/spell/trick/fleck/AbstactFleckTrick.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        entities.stream()
+            .filter(serverPlayer -> playerList.map(list -> list.contains(serverPlayer)).orElse(true)) //if a list of players was specified, filter against it
+            .forEach(serverPlayer -> serverPlayer.getComponent(ModEntityCumponents.FLECKS).addFleck(id, makeFleck(ctx, fragments)));
+
+        return fragments.getFirst();
+    }
+
+    protected abstract Fleck makeFleck(SpellContext ctx, List<Fragment> fragments) throws BlunderException, BlunderException, BlunderException;
```
Comment (StellarWitch7) [inline src/main/java/dev/enjarai/trickster/spell/trick/fleck/AbstactFleckTrick.java:60]: How even... three blunder exceptions?

File: src/main/resources/assets/trickster/lavender/entries/tome_of_tomfoolery/ploys/flecks.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Displays a spell circle at the given location.
+
+;;;;;
+
+<|glyph@trickster:templates|trick-id=trickster:draw_line,title=Artist's Plot|>
+
+number, vector, vector, [entity[] | entity] -> number
+
+Displays a line between two given position, optionally only drawing it to the given player(s). returning the given id.
```
Comment (StellarWitch7) [inline src/main/resources/assets/trickster/lavender/entries/tome_of_tomfoolery/ploys/flecks.md:27]: Remove the "optionally [...] to the given player(s)."

File: src/main/java/dev/enjarai/trickster/fleck/FleckType.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import dev.enjarai.trickster.Trickster;
+import io.wispforest.endec.StructEndec;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.SimpleRegistry;
+
+public record FleckType<T extends Fleck>(StructEndec<T> endec) {
+
+    public static final RegistryKey<Registry<FleckType<?>>> REGISTRY_KEY = RegistryKey.ofRegistry(Trickster.id("fleck_type"));
+    public static final Registry<FleckType<?>> REGISTRY = new SimpleRegistry<>(REGISTRY_KEY, Lifecycle.stable());
```
Comment (StellarWitch7) [inline src/main/java/dev/enjarai/trickster/fleck/FleckType.java:13]: Registry should be made in a fashion similar to those of Louis' recently merged PR.

**Review by StellarWitch7 — CHANGES_REQUESTED** [review]
Don't forget to load registry classes during mod initialization using an empty function.

File: src/main/java/dev/enjarai/trickster/spell/trick/fleck/AbstactFleckTrick.java (hunk 2 above)
Comment (Vlue) [reply src/main/java/dev/enjarai/trickster/spell/trick/fleck/AbstactFleckTrick.java:60]: intellij skill issue

File: src/main/java/dev/enjarai/trickster/fleck/FleckType.java (hunk 4 above)
Comment (Vlue) [reply src/main/java/dev/enjarai/trickster/fleck/FleckType.java]: made them like the spell executor type. which works i think? it builds.

**Comment by enjarai** [comment]
Just woke up, give me a bit to review it as well before merge please! ^^

**Comment by StellarWitch7** [comment]
I think @Vlue wanted you to take a look at the UVs.

**Comment by StellarWitch7** [comment]
Oh also just noticed that the abstract fleck trick has been misspelled. In addition, I see that the excess `throws` have been removed. However, there should still be one, as the fleck-making abstract method may blunder.

**Review by enjarai — APPROVED** [review]
Needs docs, but other than that, works splendidly!

**Comment by enjarai** [comment]
> I think @Vlue wanted you to take a look at the UVs.

I dont have any issue with the UVs as is. Sure, they're a bit squished near the end of the line, but that isn't an issue imo.

**Review by StellarWitch7 — APPROVED** [review]
I may have added something a little silly, for fun.

**Comment by StellarWitch7** [comment]
Some conflicts need to be resolved locally, unfortunately.

**Pull request merged**